actionview/lib/action_view/lookup_context.rb

```diff
@@ -54,24 +54,17 @@ module Accessors #:nodoc:
     register_detail(:handlers) { Template::Handlers.extensions }
 
     class DetailsKey #:nodoc:
-      alias :eql? :equal?
+      @store = Concurrent::Map.new
 
-      @details_keys = Concurrent::Map.new
+      # Make hash uniq.
+      alias_method :hash, :object_id
```

Comment:
https://github.com/rails/rails/pull/23756

> An object's hash value is not unique, so it's possible for this cache
key to produce colliding keys with no resolution. This commit move
cache on to the details key object itself, so we know that the digests
are always unique per details key object.